client/error.py

```diff
@@ -240,6 +277,71 @@ def to_text(self) -> str:
         column = click.style(str(self.column), fg="yellow")
         return f"{path}:{line}:{column} {self.description}"
 
+    def to_sarif(self) -> Dict[str, Any]:
+        return {
+            "ruleId": "PYRE-ERROR-" + str(self.code)
+            if self.code is not None
+            else "PYRE-ERROR-MDL",
+            "level": "error",
+            "message": {"text": self.description},
+            "locations": [
+                {
+                    "physicalLocation": {
+                        "artifactLocation": {
+                            "uri": str(self.path) if self.path is not None else None,
+                        },
+                        "region": {
+                            "startLine": self.line,
+                            "startColumn": self.column + 1,
+                            "endLine": self.stop_line,
+                            "endColumn": self.stop_column,
+                        },
+                    },
+                },
+            ],
+        }
+
+    def get_sarif_rule(self) -> Dict[str, Any]:
+        return {
+            "id": "PYRE-ERROR-" + str(self.code)
+            if self.code is not None
+            else "PYRE-ERROR-MDL",
+            "name": "ModelVerificationError",
+            "shortDescription": {"text": "Model verification error"},
+            "helpUri": "https://www.pyre-check.org",
+            "help": {"text": "Model Verification error"},
+        }
+
+
+def errors_to_sarif(
+    errors: Union[
+        Sequence[Error], Sequence[LegacyError], Sequence[ModelVerificationError]
+    ]
+) -> Dict[str, Any]:
+
+    results: List[Dict[str, Any]] = []
+    rules: List[Dict[str, Any]] = []
+    for error in errors:
+        results.append(error.to_sarif())
+        rules.append(error.get_sarif_rule())
```

Comment:
Cosmetic nit:
```
results = [ error.to_sarif() for error in errors ]
rules = [ error.get_sarif_rule() for error in errors ]
```

Reply:
Wouldn't that be `O(2n)` instead of `O(n)` as in the diff? :) I will make the change nonetheless